### PR TITLE
Stop pachctl mount panicking on unmount and 200x performance 😉 

### PR DESCRIPTION
**Pull request by lukemarsden**
Looks like the go-fuse lib changed its behavior from the consumer having to call Serve() to Mount() calling Serve() automatically, this change switches the extra Serve() for a Wait() so that we just wait on the lib's call. This fixes the following panic (and various flavors like it):

```
panic: sync: negative WaitGroup counter

goroutine 835 [running]:
sync.(*WaitGroup).Add(0xc000db80b8, 0xc0016c01b0)
        /snap/go/8627/src/sync/waitgroup.go:74 +0x105
sync.(*WaitGroup).Done(0x0)
        /snap/go/8627/src/sync/waitgroup.go:99 +0x25
github.com/hanwen/go-fuse/v2/fuse.(*Server).loop(0xc000db8000, 0x0)
        /home/luke/gocode/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.3/fuse/server.go:448 +0x194
github.com/hanwen/go-fuse/v2/fuse.(*Server).Serve(0xc000db8000)
        /home/luke/gocode/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.3/fuse/server.go:367 +0x35
created by github.com/hanwen/go-fuse/v2/fs.Mount
        /home/luke/gocode/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.3/fs/mount.go:32 +0x150
FAIL    github.com/pachyderm/pachyderm/v2/src/server/pfs/fuse   12.416s
```

Also, upgrade go-fuse to the latest version (although the previous version we were using had the same problem).

FInally, disable progress bars as they cause throughput to drop from 200MB/sec to 1MB/sec 😮

### CODE REVIEW

**Comment by lukemarsden** [comment]
Note: the FUSE tests don't actually seem to be running on master. They were failing.

File: src/server/pfs/fuse/fuse.go [hunk 1]
```diff
@@ -73,6 +74,11 @@ func Mount(c *client.APIClient, target string, opts *Options) (retErr error) {
 			}
 		}
 	}()
+	fmt.Println("Uploading files to Pachyderm...")
+	// Rendering progress bars for thousands of files significantly slows down
+	// throughput. Disabling progress bars takes throughput from 1MB/sec to
+	// 200MB/sec on my system, when uploading 18K small files.
+	progress.Disable()
```
Comment (albscui) [inline src/server/pfs/fuse/fuse.go:81]: Instead of hard disabling, maybe control it via options?

**Review by msteffen — APPROVED** [review]
LGTM—what a find!

File: src/server/pfs/fuse/fuse.go (hunk 1 above)
Comment (lukemarsden) [reply src/server/pfs/fuse/fuse.go:81]: I don't think we should ever give people the option to turn on something which destroys performance so badly 😄 especially not with `pachctl mount` which - as it's an interactive mount - is potentially more likely to deal with many small files.

**Comment by lukemarsden** [comment]
The failing tests are in PPS and so unrelated to this change, merging.

**Pull request merged**